src/thunderbots/software/grsim_communication/motion_controller/motion_controller.cpp

```diff
@@ -120,10 +113,24 @@ Vector MotionController::determineLinearVelocity(const Robot robot, const Point
     double new_robot_velocity_magnitude =
         std::sqrt(std::max<double>((dest - robot.position()).len() - 0.01, 0)) * 2 +
         desired_final_speed;
-    new_robot_velocity_magnitude = std::clamp<double>(new_robot_velocity_magnitude, 0,
+
+    // https://github.com/UBC-Thunderbots/Software/issues/270
+    // Check for the case where we are moving away from the target AND the additional
+    // velocity is greater in magnitude than the current velocity, as the above
+    // function will increase in magnitude as the distance to the destination increases
+    if ((robot.velocity().orientation() - unit_vector_to_dest.orientation())
+                .angleMod()
+                .abs() > Angle::quarter() &&
+        additional_velocity.len() < robot.velocity().len())
```

Comment:
Could we break this out into a few boolean variables? I think that would make this much more readable.

Reply:
broke out the first condition into a bool, 2nd condition is pretty self-explanatory

Reply:
Awesome! Could we rename `moving_in_opposite_direction` to `moving_in_opposite_direction_from_target` just to be super clear?

Reply:
renamed to `moving_away_from_dest`